Bolts/Common/BFTask.m

```diff
@@ -174,6 +174,74 @@ + (instancetype)taskForCompletionOfAllTasksWithResults:(nullable NSArray<BFTask
     }];
 }
 
++ (instancetype)taskForCompletionOfAnyTask:(nullable NSArray<BFTask *> *)tasks
+{
+    __block int32_t total = (int32_t)tasks.count;
+    if (total == 0) {
+        return [self taskWithResult:nil];
+    }
+    
+    __block int succeeded = 0;
+    __block int32_t cancelled = 0;
+    
+    NSObject *lock = [NSObject new];
+    NSMutableArray *errors = [NSMutableArray new];
+    NSMutableArray *exceptions = [NSMutableArray new];
+    
+    BFTaskCompletionSource *source = [BFTaskCompletionSource taskCompletionSource];
+    for (BFTask *task in tasks) {
+        [task continueWithBlock:^id(BFTask *task) {
+            if (task.exception != nil) {
+                @synchronized(lock) {
+                    [exceptions addObject:task.exception];
+                }
+            } else if (task.error != nil) {
+                @synchronized(lock) {
+                    [errors addObject:task.error];
+                }
+            } else if (task.cancelled) {
+                OSAtomicIncrement32Barrier(&cancelled);
+            }
+            
+            if (task.completed && !task.faulted && !task.cancelled) {
```

Comment:
I feel like you can unify this as `else` statement on the above condition. Also, there is no way this task is not going to be completed here, since `continueWithBlock:` is called only after it's completed.